### PR TITLE
Fixes that were discussed

### PR DESCRIPTION
**Pull request by hiroshihorie**
- Reset track settings on quick re-connect (resume)
- Remove `didReconnect` (confusing if quick or full)

Few things to confirm:
- [ ] Track settings syncing (sendTrackSettings) is only required for Resume (Quick reconnect) ? 

Todo:
- [x] More logging

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]

File: Sources/LiveKit/Publications/RemoteTrackPublication.swift [hunk 1]
```diff
@@ -111,36 +101,47 @@ public class RemoteTrackPublication: TrackPublication {
     ///
     /// This is useful when the participant is off screen, you may disable streaming down their video to reduce bandwidth requirements.
     @discardableResult
-    public func set(enabled: Bool) -> Promise<Void> {
+    public func set(enabled newValue: Bool) -> Promise<Void> {
```
[the remark this answers is not in the document]
Comment (davidzhao) [reply Sources/LiveKit/Publications/RemoteTrackPublication.swift:104]: what about having a synced flag? and flipping the flag if latest state has been successfully synced with server.

File: Sources/LiveKit/Publications/RemoteTrackPublication.swift [hunk 2]
```diff
@@ -164,22 +165,26 @@ public class RemoteTrackPublication: TrackPublication {
...
+    }
 
-            guard self.subscribed else {
-                throw TrackError.state(message: "Cannot modify this track setting when not subscribed")
-            }
-        }
+    var userCanModifyTrackSettings: Bool {
+
+        // adaptiveStream must be disabled and must be subscribed
+        guard let adaptiveStream = self.participant?.room.options.adaptiveStream,
```
Comment (davidzhao) [inline Sources/LiveKit/Publications/RemoteTrackPublication.swift:183]: should this be `!isAdaptiveStreamEnabled` ? it's still possible to modify audio tracks (disabling them).

File: Sources/LiveKit/Publications/RemoteTrackPublication.swift [hunk 3]
```diff
@@ -316,23 +304,33 @@ extension RemoteTrackPublication {
...
-            self?.log("[adaptiveStream] failed to send trackSettings, error: \(error)", .error)
-        }.always(on: .main) { [weak self] in
-            self?.asTimer.restart()
+        // keep old settings
+        let oldSettings = _state.trackSettings
+        // update state
+        _state.mutate { $0.trackSettings = newSettings }
+
+        // log when flipping from enabled -> disabled
+        if oldSettings.enabled, !newSettings.enabled {
```
Comment (davidzhao) [inline Sources/LiveKit/Publications/RemoteTrackPublication.swift:321]: nice!

**Review by davidzhao — COMMENTED** [review]
this looks good! thinking out loud. I'm a bit unease of using persisted `trackSettings` and keep flipping that to old state upon failure. would suggest keeping a synced flag and updating server if it hasn't been synced.

File: Sources/LiveKit/Publications/RemoteTrackPublication.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (hiroshihorie) [reply Sources/LiveKit/Publications/RemoteTrackPublication.swift:104]: Sure, I'll make adjustments.

File: Sources/LiveKit/Publications/RemoteTrackPublication.swift (hunk 2 above)
Comment (hiroshihorie) [reply Sources/LiveKit/Publications/RemoteTrackPublication.swift:183]: Got it, audio should be ok to modify.

**Pull request merged**